src/PrestaShopBundle/Twig/Component/QuickAccess.php

```diff
@@ -28,14 +28,182 @@
 
 namespace PrestaShopBundle\Twig\Component;
 
-use Link;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Adapter\QuickAccess\Repository\QuickAccessRepository;
+use PrestaShop\PrestaShop\Core\Domain\Employee\Exception\InvalidEmployeeIdException;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagSettings;
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagStateCheckerInterface;
+use PrestaShopBundle\Entity\Repository\TabRepository;
+use PrestaShopBundle\Service\DataProvider\UserProvider;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
 use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
 
 #[AsTwigComponent(template: '@PrestaShop/Admin/Component/Layout/quick_access.html.twig')]
 class QuickAccess
 {
-    public array $quickAccess;
-    public Link $link;
-    public string $quickAccessCurrentLinkIcon;
-    public string $quickAccessCurrentLinkName;
+    /**
+     * link to new product creation form
+     */
+    private const NEW_PRODUCT_LINK = 'index.php/sell/catalog/products/new';
+
+    /**
+     * link to new product creation form for product v2
+     */
+    private const NEW_PRODUCT_V2_LINK = 'index.php/sell/catalog/products-v2/create';
+
+    /**
+     * List of Quick Accesses to display.
+     */
+    private array $quickAccesses = [];
+
+    public function __construct(
+        private readonly LegacyContext $context,
+        private readonly QuickAccessRepository $quickAccessRepository,
```

Comment:
```suggestion
        private readonly QuickAccessRepositoryInterface $quickAccessRepository,
```
And you have to add the following definition:
```yaml
PrestaShop\PrestaShop\Core\QuickAccess\QuickAccessRepositoryInterface:
    alias: PrestaShop\PrestaShop\Adapter\QuickAccess\Repository\QuickAccessRepository
```
This way we can switch to another implementation easily